vhdx/src/vhdx.rs

```diff
@@ -0,0 +1,73 @@
+mod vhdx_header;
+mod vhdx_metadata;
+
+use crate::vhdx_header::{RegionTableEntry, VhdxHeader};
+
+use std::collections::btree_map::BTreeMap;
+use std::fmt::{self, Display};
+use std::fs::File;
+
+pub enum Error {
+    VhdxHeaderParsingFailed(vhdx_header::Error),
+    VhdxRegionEntryParsingFailed(vhdx_header::Error),
+}
+
+pub type Result<T> = std::result::Result<T, Error>;
+
+impl Display for Error {
+    #[remain::check]
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        use self::Error::*;
+
+        #[sorted]
+        match self {
+            VhdxHeaderParsingFailed(e) => write!(f, "Failed to parse VHDx Header {}", e),
+            VhdxRegionEntryParsingFailed(e) => write!(f, "Failed to parse VHDx Header {}", e),
+        }
+    }
+}
+
+const REGION_SIZE: u64 = 64 * 1024; // Each header size if 64KB
+const REGION1_START: u64 = 192 * 1024; // Region 1 start in Bytes
+const REGION2_START: u64 = 256 * 1024; // Region 2 start in Bytes
+const REGION1_ENTRY_START: u64 = REGION1_START + REGION_SIZE;
+const REGION2_ENTRY_START: u64 = REGION2_START + REGION_SIZE;
+
+pub struct Vhdx {
+    vhdx_header: VhdxHeader,
+    region_entries: BTreeMap<u64, u64>,
+    bat_entry: RegionTableEntry,
+    mdr_entry: RegionTableEntry,
+}
+
+impl Vhdx {
+    pub fn new(f: &mut File) -> Result<Vhdx> {
+        let vhdx_header = match VhdxHeader::new(f) {
+            Ok(vhdx_header) => vhdx_header,
+            Err(e) => {
+                return Err(Error::VhdxHeaderParsingFailed(e));
+            }
+        };
```

Comment:
This can be simplified with the following form:
```suggestion
        let vhdx_header = VhdxHeader::new(f).map_err(Error::VhdxHeaderParsingFailed)?;
```